### PR TITLE
Create Unit tests for SpringBoot controller/DAO/service

### PR DESCRIPTION
**Pull request by AJun01**

### CODE REVIEW

File: code/backend/course-builder-chatbot-ms/pom.xml [hunk 1]
```diff
@@ -13,50 +12,27 @@
...
-		<url/>
-	</scm>
+
 	<properties>
 		<java.version>17</java.version>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<maven.compiler.source>17</maven.compiler.source>
+		<maven.compiler.target>17</maven.compiler.target>
 	</properties>
-	<dependencies>
```
Comment (battalcevik) [inline code/backend/course-builder-chatbot-ms/pom.xml:32]: you have to add back these dependencies on pom.xml file

**Review by battalcevik — CHANGES_REQUESTED** [review]
You need to add all dependencies back to pom.xml file please

**Review by ananyasingh7 — CHANGES_REQUESTED** [review]
Your pom.xml is removing my dependencies!

File: code/backend/course-builder-chatbot-ms/pom.xml [hunk 2]
```diff
@@ -13,50 +12,27 @@
...
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
```
Comment (ananyasingh7) [inline code/backend/course-builder-chatbot-ms/pom.xml:34]: DO NOT REMOVE THIS!

**Review by ananyasingh7 — CHANGES_REQUESTED** [review]

**Comment by igotnowifi-official** [comment]
For Trevor: to review for iteration 3, please ignore for iteration 2 submission. Thank you!

File: code/backend/course-builder-chatbot-ms/pom.xml [hunk 3]
```diff
@@ -97,15 +97,24 @@
 			<artifactId>jackson-databind</artifactId>
 		</dependency>
 
-		<dependency>
```
Comment (ananyasingh7) [inline code/backend/course-builder-chatbot-ms/pom.xml:100]: Why is this being deleted?

File: code/backend/course-builder-chatbot-ms/pom.xml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (ananyasingh7) [inline code/backend/course-builder-chatbot-ms/pom.xml:3]: Let's not format this, keep it the same

**Review by ananyasingh7 — CHANGES_REQUESTED** [review]

File: code/backend/course-builder-chatbot-ms/pom.xml (hunk 3 above)
Comment (AJun01) [reply code/backend/course-builder-chatbot-ms/pom.xml:100]: it was duplicated, there is a same section at line 33

**Pull request closed without merging**